plugins/outputs/groundwork/slog/slog.go

```diff
@@ -0,0 +1,111 @@
+package slog
+
+import (
+	"context"
+	"encoding/json"
+	"log/slog"
+	"strings"
+	"sync"
+
+	"github.com/influxdata/telegraf"
+)
+
+// NewLogger creates telegraf.Logger adapter for slog.Logger
+func NewLogger(l telegraf.Logger) *slog.Logger {
+	return slog.New(&TlgHandler{Log: l})
+}
+
+// TlgHandler translates slog.Record into telegraf.Logger call
+// inspired by https://github.com/golang/example/blob/master/slog-handler-guide/README.md
+type TlgHandler struct {
+	attrs  []slog.Attr
+	groups []string
+
+	once sync.Once
+
+	GroupsFieldName  string
+	MessageFieldName string
+
+	Log telegraf.Logger
+}
+
+func (h *TlgHandler) Enabled(_ context.Context, level slog.Level) bool {
+	l := h.Log.Level()
+	switch level {
+	case slog.LevelDebug:
+		return l >= telegraf.Debug
+	case slog.LevelInfo:
+		return l >= telegraf.Info
+	case slog.LevelWarn:
+		return l >= telegraf.Warn
+	case slog.LevelError:
+		return l >= telegraf.Error
+	default:
+		return l >= telegraf.Info
+	}
+}
+
+func (h *TlgHandler) Handle(_ context.Context, r slog.Record) error {
+	h.once.Do(func() {
+		if h.GroupsFieldName == "" {
+			h.GroupsFieldName = "logger"
+		}
+		if h.MessageFieldName == "" {
+			h.MessageFieldName = "message"
+		}
+	})
```

Comment:
Why? Why not simply do this in the `New` function without any check?

Reply:
Done